erpnext/hr/doctype/appraisal/appraisal.py

```diff
@@ -15,66 +15,103 @@
 class Appraisal(Document):
 	def validate(self):
 		if not self.status:
-			self.status = "Draft"
-
-		if not self.goals:
-			frappe.throw(_("Goals cannot be empty"))
+			self.status = 'Draft'
 
 		validate_active_employee(self.employee)
 		set_employee_name(self)
 		self.validate_dates()
 		self.validate_existing_appraisal()
-		self.calculate_total()
+		self.calculate_total_for_kra_assessment()
+		self.check_for_kra_assessment()
+		self.check_for_behavioural_assessment()
+		self.check_for_self_improvement_areas()
+		self.check_for_new_designation()
 
 	def get_employee_name(self):
-		self.employee_name = frappe.db.get_value("Employee", self.employee, "employee_name")
+		self.employee_name = frappe.db.get_value('Employee', self.employee, 'employee_name')
 		return self.employee_name
 
 	def validate_dates(self):
 		if getdate(self.start_date) > getdate(self.end_date):
-			frappe.throw(_("End Date can not be less than Start Date"))
+			frappe.throw(_('End Date can not be less than Start Date'))
 
 	def validate_existing_appraisal(self):
-		chk = frappe.db.sql("""select name from `tabAppraisal` where employee=%s
+		chk = frappe.db.sql('''select name from `tabAppraisal` where employee=%s
```

Comment:
Query formatting. 

- select -> SELECT
- from -> FROM
- where-> WHERE
- and -> AND
- or -> OR